### PR TITLE
add parallel find_map

### PR DESCRIPTION
**Pull request by nohzafk**
add `iterator_find_pmap` `list_find_pmap`.

these functions are similar to stdlib `iterator.find_pmap` `list.find_pmap`, which will stop the parallel execution once the first Ok value is found.

### CODE REVIEW

**Comment by PastMoments** [comment]
Took me a while to get around to this. It looks good in general to me, I'd appreciate adding the iterator_find_pmap versions to both the README and the tests for completeness' sake.

File: README.md [hunk 1]
```diff
@@ -40,6 +43,23 @@ pub fn main() {
...
+    }
+  }
+  let list_input = list.range(0, 1000)
+
+  let parallel_result =
+    list_input
+    |> parallel_map.list_find_pmap(find_map_func, WorkerAmount(16), 100)
+
+  let sequential_result = list_input |> list.find_map(find_map_func)
+  // verify that both parallel_result and parallel_result are Ok(1002)
```
Comment (PastMoments) [inline README.md]: I'd also like to have sequential_result above parallel_result, just to have all of the examples be consistent

Comment (nohzafk) [reply README.md]: Updated

**Comment by nohzafk** [comment]
I've update the code, please review.

Also we are uinsg stdlib 0.36.0,  which has no `iterator.find_map` method, should we update to the latest 0.390 ?

**Comment by PastMoments** [comment]
> I've update the code, please review.
> 
> Also we are uinsg stdlib 0.36.0, which has no `iterator.find_map` method, should we update to the latest 0.390 ?

Yeah, we can update to 0.39 with this change
Also, please format it to pass the test.

**Comment by nohzafk** [comment]
All right, done

**Pull request merged**